disco/storage/outputs.py

```diff
@@ -0,0 +1,250 @@
+    def validate_output(self, output):
+        """Given an output, check if it contains desired reports
+        
+        Parameters
+        ----------
+        output: pathlib.Path
+
+        Returns
+        -------
+        str, the valid output directory containing tables.
+        """
+
+
+class SimulationOutput(OutputBase):
+    """Output instance class for handling Jade output"""
+    
+    @property
+    def output_type(self):
+        return OutputType.JADE
+    
+    def validate_output(self, output):
+        """Check if output containing all desired reports
+        
+        Parameters
+        ----------
+        output: pathlib.Path
+        """
+        report_files = [output / table_name for table_name in self.table_names[:-1]]
+        report_exists = [report_file.exists() for report_file in report_files]
+        if not all(report_exists):
+            raise ValueError(f"The output '{output}' does not contain valid reports.")
+        return output
+
+
+class PipelineSimulationOutput(OutputBase):
+    """Output class for handling Jade pipeline output"""
+    
+    @property
+    def output_type(self):
+        return OutputType.JADE_PIPELINE
+    
+    def validate_output(self, output):
+        """Return stage output that contains desired reports
+        
+        Parameters
+        ----------
+        output: str or pathlib.Path
+        """
+        for path in output.iterdir():
+            if not str(path.name).startswith("output-stage"):
+                continue
+            report_files = [path / table_name for table_name in self.table_names[:-1]]
+            report_exists = [report_file.exists() for report_file in report_files]
+            if all(report_exists):
+                return path
+        
+        raise IngestionError(f"No stage output in '{output}' contain valid reports")
+
+
+class DermsSimulationOutput(OutputBase):
+
+    @property
+    def output_type(self):
+        return OutputType.DERMS
+
+    @property
+    def result_file(self):
+        for path in self.output.iterdir():
+            if not path.is_dir():
+                continue
+            result_file = self.output / path / "results.json"
+            if not result_file.exists():
+                continue
+            return result_file
+        raise IngestionError("No JADE results.json file found")
+
+    @property
+    def derms_info_file(self):
+        return self.output / DERMS_INFO_FILENAME
+
+    def validate_output(self, output):
+        report_files = [output / table_name for table_name in self.table_names[:-1]]
+        report_exists = [report_file.exists() for report_file in report_files]
+        if not all(report_exists):
+            raise IngestionError(f"The output '{output}' does not contain valid reports.")
+        return output
+
+
+def get_creation_time(dir_or_file):
+    if platform.system() == "Windows":
+        return os.path.getctime(dir_or_file)
+    
+    stat = os.stat(dir_or_file)
+    try:
```

Comment:
```
if "darwin" in sys.platform:
```
?

Reply:
FreeBSD series use st_birthtime, including Mac. I changed the comment note.

Reply:
Sorry, I'll be more clear. You could explicitly check that you are on a Mac with `if "darwin" in sys.platform instead of relying on exception handling.